test/unit/math/gpu/add_test.cpp

```diff
@@ -0,0 +1,134 @@
+#ifdef STAN_OPENCL
+#include <stan/math/prim/mat.hpp>
+#include <stan/math/gpu/add.hpp>
+#include <gtest/gtest.h>
+
+TEST(MathMatrixGPU, add_v_exception_pass) {
+  stan::math::vector_d d1, d2;
+
+  d1.resize(3);
+  d2.resize(3);
+  stan::math::matrix_gpu d11(d1);
+  stan::math::matrix_gpu d22(d2);
+  stan::math::matrix_gpu d33(3, 1);
+  EXPECT_NO_THROW(d33 = stan::math::add(d11, d22));
```

Comment:
There are no tests making sure the addition is valid. Should we be checking for that since there are no direct tests of the kernel?

Reply:
I added a test with vectro, row vector and matrix additions.